### PR TITLE
Introduce Service classes to minimize code duplication

### LINKED ISSUE #829: Minimize data duplication around systemd service files used in each test
We are using very similar systemd service files in each test, so it would be great if we can reuse those service file among tests. One approach could be to create `Service` classes to represent systemd service file similarly as we represent BlueChi configuration files in `BlueChiConfig` classes.

### PR DESCRIPTION
**Pull request by mwperina**
Introduces Service and SimpleService to minimize code duplication around
systemd services used in integration tests.

Fixes: https://github.com/eclipse-bluechi/bluechi/issues/829
Signed-off-by: Martin Perina <mperina@redhat.com>

### CODE REVIEW

File: tests/bluechi_test/machine.py [hunk 1]
```diff
@@ -100,18 +101,15 @@ def systemctl_start_and_wait(self, service_name: str, sleep_after: float = 0.0)
...
-    def copy_systemd_service(self, service_file_name: str, source_dir: str = "systemd",
-                             target_dir: str = os.path.join("/", "etc", "systemd", "system")):
-        source_path = os.path.join(source_dir, service_file_name)
-        content = read_file(source_path)
-
-        LOGGER.debug(f"Copy local systemd service '{source_path}' to container path '{target_dir}'\
-             with content:\n{content}")
-        self.create_file(target_dir, service_file_name, content)
+    def install_systemd_service(self, service: Service,
+                                target_dir: str = os.path.join("/", "etc", "systemd", "system")):
```
Comment (engelmi) [inline tests/bluechi_test/machine.py:105]: NIT: Maybe we could move the `target_dir` also to the `Service` class similar to the file name (`service.name`)?

File: tests/bluechi_test/service.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+    def __str__(self) -> str:
+        with io.StringIO() as content:
+            self.cp.write(content, space_around_delimiters=False)
+            return content.getvalue()
+
+    def set_option(self, section: Section, name: Option, value: str) -> None:
+        self.cp.set(section.value, name.value, value)
+
+    def get_option(self, section: Section, name: Option) -> str:
```
Comment (engelmi) [inline tests/bluechi_test/service.py]: NIT:
```suggestion
    def get_option(self, section: Section, name: Option) -> Union[str, None]:
```

**Review by engelmi — APPROVED** [review]
LGTM
Only two NITs

File: tests/bluechi_test/service.py (hunk 2 above)
Comment (mwperina) [reply tests/bluechi_test/service.py]: Done

File: tests/bluechi_test/machine.py (hunk 1 above)
Comment (mwperina) [reply tests/bluechi_test/machine.py:105]: I've create https://github.com/eclipse-bluechi/bluechi/issues/830 to handle it

**Pull request merged**